From 4837ad0a22a7525bde788c2358ce9fe86d54c755 Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Tue, 17 May 2016 17:52:37 +0100
Subject: [PATCH] Allow longer surname fields.

I think the ub-surname-length limit of 40 only applies to the
surname field in an X.400 address (CHOICE [3] in GeneralName),
which can appear in various extensions.  For a surname occurring
in a RelativeDistinguishedName, Appendix A.1 of RFC5280 implies
that the corresponding length for an X520Name is ub-name (32768).
---
 checks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/checks.c b/checks.c
index 209d47fe840f..4bad0d194102 100644
--- a/checks.c
+++ b/checks.c
@@ -438,7 +438,7 @@ static const struct
 	{ &obj_dnQualifier, 1, ub_name, ERR_DN_QUALIFIER_SIZE }, /* Not sure */
 	{ &obj_pkcs9_emailAddress, 1, 255, ERR_EMAIL_SIZE },
 	{ &obj_givenName, 1, ub_name, ERR_GIVEN_NAME_SIZE },
-	{ &obj_surname, 1, 40, ERR_SURNAME_SIZE }
+	{ &obj_surname, 1, ub_name, ERR_SURNAME_SIZE }
 };
 
 static void CheckNameEntryValid(X509_NAME_ENTRY *ne)
-- 
2.8.0.rc3.226.g39d4020

